### PR TITLE
Add support for Lix (Nix fork)

### PR DESCRIPTION
**Pull request by 9999years**
## What does this PR do

Lix is a fork of Nix 2.18 focused on maintainability and user experience. It has a different format for the version, to distinguish it from CppNix:

    $ nix --version
    nix (Lix, like Nix) 2.91.0

This PR adjusts the Nix version parsing code to account for that, and also removes a few `panic!()`s along the way.

See: <https://lix.systems/>

## Standards checklist

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself

### CODE REVIEW

File: src/steps/os/unix.rs [hunk 1]
```diff
@@ -449,18 +451,44 @@ pub fn run_nix(ctx: &ExecutionContext) -> Result<()> {
...
+    if get_version_cmd_first_line_stdout.is_empty() {
+        return Err(eyre!("`nix --version` output was empty"));
+    }
+
+    let captures = NIX_VERSION_REGEX.captures(get_version_cmd_first_line_stdout);
+    let raw_version = match &captures {
+        None => {
+            return Err(eyre!(
+                "`nix --version` output was weird: {get_version_cmd_first_line_stdout:?}\n\
+                If the `nix --version` output format changed, please file an issue to Topgrade"
```
Comment (SteveLauC) [inline src/steps/os/unix.rs:478]: Nice!

**Comment by 9999years** [comment]
@SteveLauC I think that's OK, Lix _is_ Nix, it's just based on an older version of Nix so the CLI/version numbers have some different conventions.

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Comment by 9999years** [comment]
Hey, quick bump — could we get a release for this? Topgrade crashes every time it hits the Nix step on my machine. Thanks so much!

**Comment by SteveLauC** [comment]
> Hey, quick bump — could we get a release for this? Topgrade crashes every time it hits the Nix step on my machine. Thanks so much!

Sure, let me take a look and drop a release:)

**Comment by SteveLauC** [comment]
v16.0.2 released

**Comment by 9999years** [comment]
Thanks so much!

**Pull request merged**